## Travis tests have failed

Hey @PrajwalM2212,
Please read the following log in order to understand the failure reason.
It'll be awesome if you fix what's wrong and commit the changes.


### 1st Build

<a href="https:&#x2F;&#x2F;travis-ci.org&#x2F;coala&#x2F;documentation&#x2F;jobs&#x2F;546195658">View build log</a>


<details>
  <summary>
    <strong>
     make html SPHINXOPTS&#x3D;-W
    </strong>
  </summary>

```
sphinx-build -b html -d _build/doctrees  -W . _build/html
Running Sphinx v1.6.5
making output directory...
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 14 source files that are out of date
updating environment: 14 added, 0 changed, 0 removed
reading sources... [100%] index                                                 
looking for now-outdated files... none found
pickling environment... done
checking consistency... 
Warning, treated as error:
/home/travis/build/coala/documentation/Users/toml_config.rst:document isn't included in any toctree
Makefile:56: recipe for target 'html' failed
make: *** [html] Error 1
```

</details>


###### TravisBuddy Request Identifier: 063679c0-8fa5-11e9-b69a-57a176dcbff9

--- a/Users/coala_as_Git_Hook.rst
+++ b/Users/coala_as_Git_Hook.rst
@@ -34,6 +34,10 @@ coala to not apply any patch by itself. Or you can run specific sections with
         Documentation on how to configure coala using the coafile
         specification.
 
+    Module :doc:`coala configuration in TOML <toml_config>`
+        Documentation on how to configure coala using TOML
+        specification.
+
 .. note::
 
     If you allow coala to auto apply patches, it's recommended to add

--- a/Users/toml_config.rst
+++ b/Users/toml_config.rst
@@ -43,7 +43,7 @@ like minified files (e.g. ``*.min.js``) and backup files (e.g. ``*.orig``)::
     ignore = [ '**.min.js', '**.orig' ]
 
 Basic TOML concepts
-~~~~~~~~~~~~~~~~~~~~
+---------------------
 This part describes the basic TOML concepts required to write coala
 configuration files in TOML
 
@@ -54,7 +54,9 @@ configuration files in TOML
 - A table is a collection of key-value pairs. Use a table for specifying
   a coala section.
 
-This is an example of a coala configuration file written in TOML ::
+This is an example of a coala configuration file written in TOML
+
+.. code::
 
     [cli]
     bears = 'SpaceConsistencyBear'
@@ -74,12 +76,14 @@ that govern a section. To write coala configuration file you will
 be using TOML strings, booleans, integers and arrays as values.
 
 Section Inheritance
-~~~~~~~~~~~~~~~~~~~~
+---------------------
 coala supports section inheritance. You can define section inheritance
 by using the key ``inherits``.
 Extra values can be appended to an inherited setting using the ``appends`` key.
 
-Consider the following configuration file in TOML ::
+Consider the following configuration file in TOML
+
+.. code::
 
   [all]
   enabled = true
@@ -105,7 +109,9 @@ In the inherited sections above, ``appends`` key specifies that the value of
 ``ignore`` in the derived sections must be appended with the value of
 ``ignore`` key in the base section.
 
-This is the same file without section inheritance::
+This is the same file without section inheritance
+
+.. code::
 
   [all]
   enabled = true
@@ -129,7 +135,7 @@ Consider another example
 
 Config file in TOML
 
-::
+.. code::
 
  [all]
  a = 1
@@ -147,7 +153,7 @@ Config file in TOML
 You can use this syntax to specify multiple inheritance
 The same is coafile appears as
 
-::
+.. code::
 
  [all]
  a = 1
@@ -169,14 +175,31 @@ The same is coafile appears as
  p = 5
  q = 6
 
+.. note::
+
+   - If you want to append multiple settings then use ``appends`` as a list
+     .. code::
+
+        appends = [ 'a', 'b']
+   - If you want to inherit multiple sections use ``inherits`` as a list
+     .. code::
+
+        inherits = [ 'section1', 'section2']
+   - You can only inherit sections
+   - You can only append settings
+   - If  a setting is redefined in the inherited section then it will
+     overwritten if appends is not used.
+
 Defining Aspects and Tastes
-~~~~~~~~~~~~~~~~~~~~~~~~~~~
+----------------------------
 
 Aspects is an alternative way to configure coala. In this mode, we don't need
 to explicitly state list of bears, coala will choose it automatically based on
 requested aspects in configuration file. To run coala in this mode, we need to
 define `aspects`, `files`, `languages`, and optionally aspect tastes setting.
-See the following example::
+See the following example
+
+.. code::
 
   [all]
   files = '**'
@@ -186,23 +209,33 @@ See the following example::
   # we can define subaspect taste through its parent
   aspectname1.subaspect_taste = ['word1', 'word2', 'word3']
 
-  ['all.python']
+  [python]
   files = '**.py'
   language = 'Python'
+  inherits = 'all'
   # appending additional aspect
   appends = 'all'
   aspects = 'aspectname3'
   # excluding certain subaspect
   excludes = 'AspectName2Subaspect'
 
+For caofile users who want to write configuration in TOML
+
+- If you are using aspects  ``a:b = 'c'``  in a section named `example`
+  then replace ``a:b = 'c'`` with ``a.b = 'c'`` or
+
+  .. code::
+
+   [example.a]
+   b = 'c'
 
 For existing coala users
-~~~~~~~~~~~~~~~~~~~~~~~~~
+-------------------------
 
 In this section we will see how to convert a complex coafile into
 a configuration file in TOML
 
-coafile ::
+.. code::
 
  [all]
  files = *.py, coantlib/**/*.py, tests/**/*.py, coantbears/**/*.py, .ci/*.py
@@ -217,15 +250,12 @@ coafile ::
  bears = SpaceConsistencyBear
  language = Python
  preferred_quotation = '
-
  default_actions = **: ApplyPatchAction
 
  [all.flakes]
  # Do not set default_action to ApplyPatchAction as it may lead to some
  # required imports being removed that might result in coala behaving weirdly.
-
  default_actions = *: ShowPatchAction
-
  bears += PyUnusedCodeBear
  language = Python
  remove_all_unused_imports = true
@@ -234,26 +264,20 @@ To convert a coafile to configuration file in TOML
 
 - Enclose all string values in quotes
 - Use array notation to depict list of strings
-- Replace ``[parent_section.inherited_section]]`` with ``[inherited.section]``
+- Replace ``[parent_section.inherited_section]`` with ``[inherited.section]``
   and add ``inherits = parent_section`` as a key-value pair
 - Use ``true`` or ``false`` to specify booleans
 - Replace ``a += b`` with
-  ::
+
+.. code::
 
    a = 'b'
    appends = 'a'
 
-- If you are using aspects  ``a:b = 'c'``  in a section named `example`
-  then replace
-  ``a:b = 'c'`` with ``a.b = 'c'`` or
-  ::
-
-   [example.a]
-   b = 'c'
 
 Using the above rules we get a configuration file in TOML
 
-::
+.. code::
 
  [all]
  files = ['*.py', 'coantlib/**/*.py', 'tests/**/*.py', 'coantbears/**/*.py',
@@ -270,7 +294,6 @@ Using the above rules we get a configuration file in TOML
  bears = 'SpaceConsistencyBear'
  language = 'Python'
  preferred_quotation = '
-
  default_actions = '**: ApplyPatchAction'
 
  [flakes]
@@ -278,7 +301,6 @@ Using the above rules we get a configuration file in TOML
  # required imports being removed that might result in coala behaving weirdly.
  inherits = 'all'
  default_actions = '*: ShowPatchAction'
-
  bears = 'PyUnusedCodeBear'
  appends = 'bears'
  language = 'Python'

--- a/Users/toml_config.rst
+++ b/Users/toml_config.rst
@@ -1,0 +1,155 @@
+Writing a coala Configuration File in TOML
+======================================================
+
+This document describes how to write configuration files for
+coala in TOML format.
+
+Naming, Scope and Location
+--------------------------
+
+You can use up to three coafiles to configure your project.
+
+1. A project-wide coafile.
+2. A user-wide coafile.
+3. A system-wide coafile.
+
+Project-Wide coafile
+~~~~~~~~~~~~~~~~~~~~
+
+It is a convention that the project-wide configuration file is named
+``.coafile.toml`` and lies in the project root directory.
+If you follow this convention, simply executing ``coala -T`` from the
+project root will execute the configuration specified in that file.
+
+Settings given in the project-wide configuration file override all settings
+given by other files and can only be overridden by settings given via the
+command line interface.
+
+User-Wide and System-Wide coafile
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can place a ``.coarc.toml`` file in your home directory to set certain
+user wide settings. Those settings will automatically be taken for all
+projects executed with that user.
+
+All settings specified here override only settings given by the system
+wide configuration file which has the lowest priority. The
+``system_coafile.toml`` must lie in the coala installation directory
+and is valid for everyone using this coala installation.
+
+It can be used to define the type of files you usually don't want to lint,
+like minified files (e.g. ``*.min.js``) and backup files (e.g. ``*.orig``)::
+
+    ignore = [ '**.min.js', '**.orig' ]
+
+Basic TOML concepts
+---------------------------------
+This part describes the basic TOML concepts required to write coala
+configuration files
+
+- TOML is case sensitive. So remember to not have duplicate sections/tables
+  or duplicate keys in same section.
+- key-value pairs are building blocks of a TOML document. Use key-value
+  pairs to specify rules for coala bears.
+- A table is a collection of key-value pairs. Use a table for specifying
+  a coala section.
+
+::
+
+    [cli]
+    bears = 'SpaceConsistencyBear'
+    files = 'src/*.c'
+    use_spaces = true
+
+    [invalidlinks]
+    enabled = false
+    files = [ '**/*.rst', 'README.rst']
+    ignore = 'venv/**'
+    bears = 'InvalidLinkBear'
+
+Here tables ``cli`` and ``invalidlinks`` are coala sections.
+The contents of the tables like ``bears``, ``files`` are rules
+that govern a section. In coala you will be using TOML strings,
+booleans, integers and arrays as values.
+
+Section Inheritance
+----------------------------
+coala supports section inheritance. You can define section inheritance
+explicitly by naming a section in the format ``["basesection.newsection"]``.
+Extra values can be appended to an inherited setting using the ``appends`` key.
+
+.. note::
+
+    In ``["basesection.newsection"]``, the quotes insides the square braces are
+    necessary for specifying section inheritance in TOML.
+
+
+Consider the following coafile::
+
+  [all]
+  enabled = true
+  overridable = 2
+  ignore = 'vendor1/'
+
+  ["all.section1"]
+  overridable = 3
+  appends = 'ignore'
+  ignore = 'vendor2/'
+  other = 'some_value'
+
+  ["all.section2"]
+  overridable = 4
+  ignore = 'vendor3/'
+  appends = 'ignore'
+  other = 'some_other_value'
+
+
+In the inherited sections above, ``appends`` key specifies that the value of
+``ignore`` in the derived sections must be appended with the value of
+``ignore`` key in the base section. This is the same file without section
+inheritance::
+
+  [all]
+  enabled = true
+  overridable = 2
+  ignore = 'vendor1/'
+
+  [section1]
+  enabled = true
+  overridable = 3
+  ignore = ['vendor1/', 'vendor2/']
+  other = 'some_value'
+
+  [section2]
+  enabled = true
+  overridable = 4
+  ignore = ['vendor1/', 'vendor3/']
+  other = 'some_other_value'
+
+
+Defining Aspects and Tastes
+---------------------------
+
+Aspects is an alternative way to configure coala. In this mode, we don't need
+to explicitly state list of bears, coala will choose it automatically based on
+requested aspects in configuration file. To run coala in this mode, we need to
+define `aspects`, `files`, `languages`, and optionally aspect tastes setting.
+See the following example::
+
+  [all]
+  files = '**'
+  aspects = ['aspectname1', 'AspectName2'] # case-insensitive
+  # defining an aspect's taste
+  aspectname1.aspect_taste = 80
+  # we can define subaspect taste through its parent
+  aspectname1.subaspect_taste = ['word1', 'word2', 'word3']
+
+  ['all.python']
+  files = '**.py'
+  language = 'Python'
+  # appending additional aspect
+  appends = 'all'
+  aspects = 'aspectname3'
+  # excluding certain subaspect
+  excludes = 'AspectName2Subaspect'
+

--- a/Users/toml_config.rst
+++ b/Users/toml_config.rst
@@ -7,14 +7,14 @@ coala in TOML format.
 Naming, Scope and Location
 --------------------------
 
-You can use up to three coafiles to configure your project.
+You can use up to three configuration files to configure your project.
 
-1. A project-wide coafile.
-2. A user-wide coafile.
-3. A system-wide coafile.
+1. A project-wide configuration file.
+2. A user-wide configuration file.
+3. A system-wide configuration file.
 
-Project-Wide coafile
-~~~~~~~~~~~~~~~~~~~~
+Project-Wide configuration file
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 It is a convention that the project-wide configuration file is named
 ``.coafile.toml`` and lies in the project root directory.
@@ -25,8 +25,8 @@ Settings given in the project-wide configuration file override all settings
 given by other files and can only be overridden by settings given via the
 command line interface.
 
-User-Wide and System-Wide coafile
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+User-Wide and System-Wide configuration file
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 You can place a ``.coarc.toml`` file in your home directory to set certain
 user wide settings. Those settings will automatically be taken for all
@@ -43,9 +43,9 @@ like minified files (e.g. ``*.min.js``) and backup files (e.g. ``*.orig``)::
     ignore = [ '**.min.js', '**.orig' ]
 
 Basic TOML concepts
----------------------------------
+~~~~~~~~~~~~~~~~~~~~
 This part describes the basic TOML concepts required to write coala
-configuration files
+configuration files in TOML
 
 - TOML is case sensitive. So remember to not have duplicate sections/tables
   or duplicate keys in same section.
@@ -54,7 +54,7 @@ configuration files
 - A table is a collection of key-value pairs. Use a table for specifying
   a coala section.
 
-::
+This is an example of a coala configuration file written in TOML ::
 
     [cli]
     bears = 'SpaceConsistencyBear'
@@ -67,38 +67,35 @@ configuration files
     ignore = 'venv/**'
     bears = 'InvalidLinkBear'
 
+
 Here tables ``cli`` and ``invalidlinks`` are coala sections.
 The contents of the tables like ``bears``, ``files`` are rules
-that govern a section. In coala you will be using TOML strings,
-booleans, integers and arrays as values.
+that govern a section. To write coala configuration file you will
+be using TOML strings, booleans, integers and arrays as values.
 
 Section Inheritance
-----------------------------
+~~~~~~~~~~~~~~~~~~~~
 coala supports section inheritance. You can define section inheritance
-explicitly by naming a section in the format ``["basesection.newsection"]``.
+by using the key ``inherits``.
 Extra values can be appended to an inherited setting using the ``appends`` key.
 
-.. note::
-
-    In ``["basesection.newsection"]``, the quotes insides the square braces are
-    necessary for specifying section inheritance in TOML.
-
-
-Consider the following coafile::
+Consider the following configuration file in TOML ::
 
   [all]
   enabled = true
   overridable = 2
   ignore = 'vendor1/'
 
-  ["all.section1"]
+  [section1]
   overridable = 3
+  inherits = 'all'
   appends = 'ignore'
   ignore = 'vendor2/'
   other = 'some_value'
 
-  ["all.section2"]
+  [section2]
   overridable = 4
+  inherits = 'all'
   ignore = 'vendor3/'
   appends = 'ignore'
   other = 'some_other_value'
@@ -106,8 +103,9 @@ Consider the following coafile::
 
 In the inherited sections above, ``appends`` key specifies that the value of
 ``ignore`` in the derived sections must be appended with the value of
-``ignore`` key in the base section. This is the same file without section
-inheritance::
+``ignore`` key in the base section.
+
+This is the same file without section inheritance::
 
   [all]
   enabled = true
@@ -127,8 +125,52 @@ inheritance::
   other = 'some_other_value'
 
 
+Consider another example
+
+Config file in TOML
+
+::
+
+ [all]
+ a = 1
+ b = 2
+
+ [java]
+ c = 3
+ d = 4
+
+ [python]
+ p = 5
+ q = 6
+ inherits = [ 'all', 'java']
+
+You can use this syntax to specify multiple inheritance
+The same is coafile appears as
+
+::
+
+ [all]
+ a = 1
+ b = 2
+
+ [java]
+ c = 3
+ d = 4
+
+ [all.python]
+ a = 1
+ b = 2
+ p = 5
+ q = 6
+
+ [java.python]
+ c = 3
+ d = 4
+ p = 5
+ q = 6
+
 Defining Aspects and Tastes
----------------------------
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Aspects is an alternative way to configure coala. In this mode, we don't need
 to explicitly state list of bears, coala will choose it automatically based on
@@ -152,4 +194,93 @@ See the following example::
   aspects = 'aspectname3'
   # excluding certain subaspect
   excludes = 'AspectName2Subaspect'
+
+
+For existing coala users
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this section we will see how to convert a complex coafile into
+a configuration file in TOML
+
+coafile ::
+
+ [all]
+ files = *.py, coantlib/**/*.py, tests/**/*.py, coantbears/**/*.py, .ci/*.py
+ max_line_length = 80
+ use_spaces = True
+
+ [all.python]
+ # Patches may conflict with autopep8 so putting them in own section so they
+ # will be executed sequentially; also we need the LineLengthBear to double
+ # check the line length because PEP8Bear sometimes isn't able to correct the
+ # linelength.
+ bears = SpaceConsistencyBear
+ language = Python
+ preferred_quotation = '
+
+ default_actions = **: ApplyPatchAction
+
+ [all.flakes]
+ # Do not set default_action to ApplyPatchAction as it may lead to some
+ # required imports being removed that might result in coala behaving weirdly.
+
+ default_actions = *: ShowPatchAction
+
+ bears += PyUnusedCodeBear
+ language = Python
+ remove_all_unused_imports = true
+
+To convert a coafile to configuration file in TOML
+
+- Enclose all string values in quotes
+- Use array notation to depict list of strings
+- Replace ``[parent_section.inherited_section]]`` with ``[inherited.section]``
+  and add ``inherits = parent_section`` as a key-value pair
+- Use ``true`` or ``false`` to specify booleans
+- Replace ``a += b`` with
+  ::
+
+   a = 'b'
+   appends = 'a'
+
+- If you are using aspects  ``a:b = 'c'``  in a section named `example`
+  then replace
+  ``a:b = 'c'`` with ``a.b = 'c'`` or
+  ::
+
+   [example.a]
+   b = 'c'
+
+Using the above rules we get a configuration file in TOML
+
+::
+
+ [all]
+ files = ['*.py', 'coantlib/**/*.py', 'tests/**/*.py', 'coantbears/**/*.py',
+         '.ci/*.py']
+ max_line_length = 80
+ use_spaces = true
+
+ [python]
+ # Patches may conflict with autopep8 so putting them in own section so they
+ # will be executed sequentially; also we need the LineLengthBear to double
+ # check the line length because PEP8Bear sometimes isn't able to correct the
+ # linelength.
+ inherits = 'all'
+ bears = 'SpaceConsistencyBear'
+ language = 'Python'
+ preferred_quotation = '
+
+ default_actions = '**: ApplyPatchAction'
+
+ [flakes]
+ # Do not set default_action to ApplyPatchAction as it may lead to some
+ # required imports being removed that might result in coala behaving weirdly.
+ inherits = 'all'
+ default_actions = '*: ShowPatchAction'
+
+ bears = 'PyUnusedCodeBear'
+ appends = 'bears'
+ language = 'Python'
+ remove_all_unused_imports = true
 

--- a/index.rst
+++ b/index.rst
@@ -27,6 +27,7 @@ Welcome to the coala documentation!
    Installing coala <Users/Install>
    Getting Started with coala <Users/Tutorial>
    Writing a coala Configuration File (coafile and coarc) <Users/coafile>
+   Writing a Configuration File in TOML <Users/toml_config>
    Generating a .coafile using coala-quickstart <Users/coala_quickstart.rst>
    Using Glob Patterns <Users/Glob_Patterns>
    Exit Codes <Users/Exit_Codes>

